### DISCUSSION TITLE
Use the Secrets Gradle plugin to store and access project secrets

### DISCUSSION THREAD

**Discussion by ryan-conway** [nimblehq/android-templates#418] [RFCs]
## Issue

[Original Thread](https://nimble-co.slack.com/archives/G41UCTGHL/p1658832220423489)

Secret management is a valid concern for all projects, with Google recommending to _not_ check in any API keys to version control. 
We can use the [Secrets Gradle plugin](https://developers.google.com/maps/documentation/android-sdk/secrets-gradle-plugin) to store and access these keys, and also make use of [build-variant specific properties](https://github.com/google/secrets-gradle-plugin#build-variant-specific-properties) so we can keep the specific keys for staging or release separately.

## Solution

- Integrate the Secrets Gradle plugin to `template-xml` and `template-compose`
- Update sample folders to reflect template changes

## Who Benefits?

Developers

## What's Next?

- [ ] Create issue to integrate Secrets Gradle plugin
- [ ] Create issue to update sample folders

**Comment by toby-thanathip** [comment]
Voted, let's add it. ✅ 

However, which key will we use for showcasing our sample projects?  As of now, we currently don't have any API keys 💭

**Reply by toby-thanathip** [reply]
> However, which key will we use for showcasing our sample projects? As of now, we currently don't have any API keys 💭

This RFC will be influenced by https://github.com/nimblehq/android-templates/discussions/434 💡

**Comment by luongvo** [comment]
Voted 👍

**Comment by chornerman** [comment]
Voted 🚀

**Comment by manh-t** [comment]
Voted 👍
